### PR TITLE
fix(router): redirects use the to query string

### LINKED ISSUE #23136: bug(router): route-redirect does not honor the query string
# Bug Report

**Ionic version:**

[ ] **4.x**
[x] **5.6**

**Current behavior:**

```
<ion-route-redirect from="/redirect" to="/foo?src=redirect"></ion-route-redirect>
```

would redirect to `/foo`

**Expected behavior:**

```
<ion-route-redirect from="/redirect" to="/foo?src=redirect"></ion-route-redirect>
```

should redirect to `/foo?src=redirect`

**Steps to reproduce:**
**Related code:**

https://runkit.com/vicb/ionic-router-redirect---query-string

using this runkit:
- click on "go to foo", the url changes to `/foo?src=button`,
- click on "redirect to foo", the url changes to `/foo` but we want `/foo?src=redirect`

**Other information:**

This feature is missing from the ionic source code.
It would be easy to implement and I am happy to contribute a fix.
I would make sense to merge #23123 first though.

### PR DESCRIPTION
**Pull request by vicb**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Build (`npm run build`) was run locally and any changes were pushed
- [x] Lint (`npm run lint`) has passed locally and any fixes were made for failures

## Pull request type

Please check the type of change your PR introduces:
- [x] Bugfix
- [ ] Feature
- [x] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?

Issue Number: fixes #23136

## What is the new behavior?

when redirect to `/some/path?qs` the query string was not set before this PR.

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

This PR also adds some comments and renaming in order to make the code easier to understand.

### CODE REVIEW

File: core/src/components/router/utils/matching.ts [hunk 1]
```diff
@@ -1,12 +1,17 @@
 import { RouteChain, RouteID, RouteRedirect } from './interface';
 
-export const matchesRedirect = (input: string[], route: RouteRedirect): route is RouteRedirect => {
-  const { from, to } = route;
+// Returns whether the given redirect matches the given path segments.
+//
+// A redirect matches when the segments of the path and redirect.from are equal.
+// Note that segments are only check until redirect.from contains a '*' which matches any path segments.
```
Comment (liamdebeasi) [inline core/src/components/router/utils/matching.ts:6]:
```suggestion
// Note that segments are only checked until redirect.from contains a '*' which matches any path segment.
```

**Review by liamdebeasi — APPROVED** [review]
Looks great!

**Comment by vicb** [comment]
Thanks @liamdebeasi 
I have committed your suggestion

**Comment by liamdebeasi** [comment]
Merged. Thank you!

**Comment by vicb** [comment]
Thanks,
I'll work on fixing #22944 and #23135 next week.

**Pull request merged**